Lib/concurrent/futures/process.py

```diff
@@ -601,6 +607,7 @@ def __init__(self, max_workers=None, mp_context=None,
         # Shutdown is a two-step process.
         self._shutdown_thread = False
         self._shutdown_lock = threading.Lock()
+        self._idle_worker_semaphore = mp_context.Semaphore(0)
```

Comment:
Since all semaphore operations seem to be happening in the same process, a `threading.Semaphore` should be sufficient (and lighter).

Reply:
Agreed, good point. A previous implementation I was trying required a multiprocessing semaphore, but this version doesn't.